app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,25 @@
+class MessagesController < ApplicationController
+  before_action :set_search
```

Comment:
no need to call set_search here, as it is already executed via before_action in ApplicationController